src/test/java/nextstep/subway/line/application/LineServiceTest.java

```diff
@@ -0,0 +1,92 @@
+package nextstep.subway.line.application;
+
+import nextstep.subway.line.domain.Line;
+import nextstep.subway.line.dto.LineDto;
+import nextstep.subway.line.repository.LineRepository;
+import nextstep.subway.station.domain.Station;
+import nextstep.subway.station.repository.StationRepository;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.transaction.annotation.Transactional;
+
+import static nextstep.subway.station.application.StationServiceTest.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
+class LineServiceTest {
+    @Autowired
+    private StationRepository stationRepository;
+
+    @Autowired
+    private LineRepository lineRepository;
+
+    @Test
+    @Transactional
+    void saveLine() {
+        Station station1 = stationRepository.save(강남역.toStation());
+        Station station2 = stationRepository.save(역삼역.toStation());
+        LineDto.CreateRequest 이호선 = new LineDto.CreateRequest()
+                .setColor("green")
+                .setName("2호선")
+                .setDistance(10);
+
+        Line expect = lineRepository.save(이호선.of(station1, station2));
+        assertThat(lineRepository.findById(expect.getId()).get()).isEqualTo(expect);
+    }
+
+    @Test
+    @Transactional
+    void findAllLines() {
+        Station station1 = stationRepository.save(강남역.toStation());
+        Station station2 = stationRepository.save(역삼역.toStation());
+        LineDto.CreateRequest 이호선 = new LineDto.CreateRequest()
+                .setColor("green")
+                .setName("2호선")
+                .setDistance(10);
+
+        Station station3 = stationRepository.save(교대역.toStation());
+        Station station4 = stationRepository.save(마두역.toStation());
+        LineDto.CreateRequest 삼호선 = new LineDto.CreateRequest()
+                .setColor("orange")
+                .setName("3호선")
+                .setDistance(10);
+
+        lineRepository.save(이호선.of(station1, station2));
+        lineRepository.save(삼호선.of(station1, station2));
+
+        lineRepository.findAll();
+        assertThat(lineRepository.findAll()).hasSize(2);
+    }
+
+    @Test
+    @Transactional
+    void deleteLineById() {
+        Station station1 = stationRepository.save(강남역.toStation());
+        Station station2 = stationRepository.save(역삼역.toStation());
+        LineDto.CreateRequest 이호선 = new LineDto.CreateRequest()
+            .setColor("green")
+            .setName("2호선")
+            .setDistance(10);
+
+        Line expect = lineRepository.save(이호선.of(station1, station2));
+        lineRepository.delete(expect);
+        assertThat(lineRepository.findById(expect.getId()).isPresent()).isFalse();
```

Comment:
`given / when / then` 패턴을 고려 해보았을때,
테스트 코드의 흐름을 명확하게 구분할 것 같아요!
- 준비과정
- 검증하고자 하는 메소드 
- assert 를 통한 검증

그리고, 이름을 한글로 사용하므로써 가독성을 좋게 만들것 같아요!(물론 이관점은 개인적인 생각입니다 ^^)


```suggestion
       Station station1 = stationRepository.save(강남역.toStation());
        Station station2 = stationRepository.save(역삼역.toStation());
        LineDto.CreateRequest 이호선_요청_파라미터 = new LineDto.CreateRequest()
            .setColor("green")
            .setName("2호선")
            .setDistance(10);
        Line 이호선 = lineRepository.save(이호선_요청_파라미터.of(station1, station2));

        lineRepository.delete(이호선);
        
        assertThat(lineRepository.findById(expect.getId()).isPresent()).isFalse();
```

Reply:
추가적으로,

`LineServiceTest` 인데 `JPARepository` 테스트 인것 같아요 ㅜ
삭제에 대한 테스트라면, `LineService.deleteLineById` 에 대한 테스트가 맞지 않을까 생각되어요 :thinking:

Reply:
😭 하다보니 로직을 바탕으로한 JPARepository가 되어버렸네요 ㅠㅠ